jetty-websocket/jetty-websocket-server/src/main/java/org/eclipse/jetty/websocket/server/JettyServerFrameHandlerFactory.java

```diff
@@ -42,4 +66,16 @@ public FrameHandler newFrameHandler(Object websocketPojo, ServletUpgradeRequest
         return super.newJettyFrameHandler(websocketPojo, new UpgradeRequestAdapter(upgradeRequest), new UpgradeResponseAdapter(upgradeResponse),
             new CompletableFuture<>());
     }
+
+    @Override
+    public void lifeCycleStopping(LifeCycle context)
+    {
+        ContextHandler contextHandler = (ContextHandler) context;
```

Comment:
I really dislike "LifeCycle" stop as a choice for this behavior.
It should be part of Context Destroyed instead.

Reply:
I think the lifecycle listener is the correct one, as this is being done for the managed status of the bean rather than anything specific to the context